tests/tests/image_atomics/mod.rs

```diff
@@ -0,0 +1,168 @@
+            label: None,
+            bind_group_layouts: &[&bind_group_layout],
+            push_constant_ranges: &[],
+        });
+    let shader = ctx.device.create_shader_module(desc);
+    let pipeline = ctx
+        .device
+        .create_compute_pipeline(&wgpu::ComputePipelineDescriptor {
+            label: Some("image atomics pipeline"),
+            layout: Some(&pipeline_layout),
+            module: &shader,
+            entry_point: Some("cs_main"),
+            compilation_options: wgpu::PipelineCompilationOptions::default(),
+            cache: None,
+        });
+
+    let tex = ctx.device.create_texture(&wgpu::TextureDescriptor {
+        label: None,
+        dimension: wgpu::TextureDimension::D2,
+        size,
+        format,
+        usage: wgpu::TextureUsages::STORAGE_BINDING
+            | wgpu::TextureUsages::STORAGE_ATOMIC
+            | wgpu::TextureUsages::COPY_SRC,
+        mip_level_count: 1,
+        sample_count: 1,
+        view_formats: &[],
+    });
+    let view = tex.create_view(&wgpu::TextureViewDescriptor {
+        format: Some(format),
+        aspect: wgpu::TextureAspect::All,
+        ..wgpu::TextureViewDescriptor::default()
+    });
+    let bind_group = ctx.device.create_bind_group(&wgpu::BindGroupDescriptor {
+        label: None,
+        layout: &pipeline.get_bind_group_layout(0),
+        entries: &[wgpu::BindGroupEntry {
+            binding: 0,
+            resource: wgpu::BindingResource::TextureView(&view),
+        }],
+    });
+
+    let mut encoder = ctx
+        .device
+        .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+    let mut rpass = encoder.begin_compute_pass(&wgpu::ComputePassDescriptor {
+        label: None,
+        timestamp_writes: None,
+    });
+    rpass.set_pipeline(&pipeline);
+    rpass.set_bind_group(0, Some(&bind_group), &[]);
+    rpass.dispatch_workgroups(size.width, size.height, 1);
+    drop(rpass);
+    ctx.queue.submit(Some(encoder.finish()));
+
+    let read_buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+        label: None,
+        size: (size.height * size.width * size.depth_or_array_layers * pixel_bytes) as u64,
+        usage: wgpu::BufferUsages::MAP_READ | wgpu::BufferUsages::COPY_DST,
+        mapped_at_creation: false,
+    });
+
+    let mut encoder = ctx
+        .device
+        .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+
+    encoder.copy_texture_to_buffer(
+        wgpu::TexelCopyTextureInfo {
+            texture: &tex,
+            mip_level: 0,
+            origin: wgpu::Origin3d::ZERO,
+            aspect: wgpu::TextureAspect::All,
+        },
+        wgpu::TexelCopyBufferInfo {
+            buffer: &read_buffer,
+            layout: wgpu::TexelCopyBufferLayout {
+                offset: 0,
+                bytes_per_row: Some(size.width * pixel_bytes),
+                rows_per_image: Some(size.height),
+            },
+        },
+        size,
+    );
```

Comment:
There's a helper struct in wgpu_test called `ReadbackBuffers` which can pull down textures and deal with padding and such if you want to make the size not a multiple of the alignment.